// @aws-sdk/client-sagemaker-runtime/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { SageMakerRuntimeServiceException as __BaseException } from "./SageMakerRuntimeServiceException";
export class InternalDependencyException extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalDependencyException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalDependencyException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalDependencyException.prototype);
        this.Message = opts.Message;
    }
}
export class InternalFailure extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalFailure",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalFailure";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalFailure.prototype);
        this.Message = opts.Message;
    }
}
export class ModelError extends __BaseException {
    constructor(opts) {
        super({
            name: "ModelError",
            $fault: "client",
            ...opts,
        });
        this.name = "ModelError";
        this.$fault = "client";
        Object.setPrototypeOf(this, ModelError.prototype);
        this.Message = opts.Message;
        this.OriginalStatusCode = opts.OriginalStatusCode;
        this.OriginalMessage = opts.OriginalMessage;
        this.LogStreamArn = opts.LogStreamArn;
    }
}
export class ModelNotReadyException extends __BaseException {
    constructor(opts) {
        super({
            name: "ModelNotReadyException",
            $fault: "client",
            ...opts,
        });
        this.name = "ModelNotReadyException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ModelNotReadyException.prototype);
        this.Message = opts.Message;
    }
}
export class ServiceUnavailable extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceUnavailable",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceUnavailable";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceUnavailable.prototype);
        this.Message = opts.Message;
    }
}
export class ValidationError extends __BaseException {
    constructor(opts) {
        super({
            name: "ValidationError",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationError";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationError.prototype);
        this.Message = opts.Message;
    }
}
export const InvokeEndpointInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Body && { Body: SENSITIVE_STRING }),
    ...(obj.CustomAttributes && { CustomAttributes: SENSITIVE_STRING }),
});
export const InvokeEndpointOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Body && { Body: SENSITIVE_STRING }),
    ...(obj.CustomAttributes && { CustomAttributes: SENSITIVE_STRING }),
});
export const InvokeEndpointAsyncInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.CustomAttributes && { CustomAttributes: SENSITIVE_STRING }),
});
export const InvokeEndpointAsyncOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
